### DISCUSSION TITLE
Memory leak while using Streamlink

### DISCUSSION THREAD

**Discussion by nigberg** [streamlink/streamlink#6063] [Q&A]
### Description

Hi, I'm using Streamlink on newly installed Windows 11, and while using the following issue occurs:
![photo_5776035237464228728_y](https://github.com/streamlink/streamlink/assets/13922881/ece824ec-bee3-4be1-b0e8-a8f8f0455e3d)
The 'Page Table' and 'Unused' are growing fastly and in about 4 hours all RAM memory is almost full.
There's no process running on my computer that uses a lot of memory - [link to video](https://www.youtube.com/watch?v=PzhfClZgEIE)
On previously installed Windows 10 it worked normally, without this issue.
I'd appreciate your help.

This is how i start the program:
```
@echo off
set link=https://twitch.tv/
set name=
set quality=best
set dir=S:\Records
set /p tok=<C:\Users\ilya\Desktop\oken.txt 
set /p tok=<C:\Users\ilya\Desktop\oken.txt 
set seconds=30
:loop
cls
echo Record %name%
echo ===========================================
set datetime=%date:~6,4%-%date:~3,2%-%date:~0,2%_%time:~0,2%-%time:~3,2%-%time:~6,2%
set datetime=%datetime: =0%
set file=%name%_%datetime%.ts
title Record - %file%
streamlink --stream-segment-thread 3 %link% %quality% -o "%dir%\%file%" --twitch-api-header="Authorization=OAuth %tok%" --twitch-api-header="Client-Id=%tok%" --twitch-api-header="Client-Integrity=%tok%" --twitch-api-header="X-Device-Id=%tok%" --twitch-api-header="Authorization=OAuth %tok%" --twitch-api-header="Client-Id=%tok%" --twitch-api-header="Client-Integrity=%tok%" --twitch-api-header="X-Device-Id=%tok%" --twitch-disable-ads
timeout %seconds%
goto loop
```

Thanks in advance!

**Comment by bastimeyer** [comment]
> There's no process running on my computer that uses a lot of memory
> On previously installed Windows 10 it worked normally, without this issue.

You are running **hundreds of Python/Streamlink processes**, which you are clearly showning in your video.

Both your screenshot and video don't say anything at all because they are not showing **details about a single process**, which is what would matter in case of an actual memory leak where the Python process keeps allocating memory infinitely. Don't be surprised if your system runs out of memory if you're launching many processes at the same time.

If there was an actual memory leak, then this wouldn't just affect a single system but every user, which is clearly not the case. There's nothing specific to Windows or Windows 11 in Streamlink's Python code and its various implementations. There were memory leaks in the past, but these were caused by a bad design in regards to the session and plugin initialization, but this has been fixed a long time ago, and it was also not relevant for regular CLI usage.

Also, the way you're trying to record and open Twitch streams is complete nonsense, with the same "token" being used for a client OAuth token, Client-ID, Client-Integrity and Device ID, and all of them are apparently set multiple times. None of that makes any sense at all.
